### DISCUSSION TITLE
add a backup cron job

### DISCUSSION THREAD

**Discussion by SQLJames** [SQLJames/factorio-server-charts#34] [Ideas]
It would be nice to have a built in cron job that can backup the data to another location. 

- SCP to another server 
- Or another PVC that users created outside of the helm chart. 

This would allow them to roll back if needed, for example if they did not create a proper backup strategy for the pvc or their retention policy on their storageclass is set to delete

**Comment by proegssilb** [comment]
Either automatically inserting a timestamp or allowing templating in the uploaded file name would probably be a good idea. Plus, there was some functionality I was thinking about for a companion app that would benefit from having timestamped saves going back the history of the game.

**Comment by digifuchsi** [comment]
I have written a simple cronjob to upload the contents of /factorio/saves to an webDAV server: https://github.com/digifuchsi/factorio-server-charts/tree/feat/backup
It is just a first draft right now to have my idea written down somewhere and needs to be tested and polished. Therefore I have not created an PR right now.

The cronjob does not keep a history of saves right now. I will implement this when I have some more time to spend on it. Right now it just suits my basic needs of not loosing my game when something bad happens with my K8s.
